src/components/editor-page/sidebar/delete-note-sidebar-entry.tsx

```diff
@@ -4,32 +4,52 @@
  * SPDX-License-Identifier: AGPL-3.0-only
  */
 
-import React, { Fragment, useState } from 'react'
+import React, { Fragment, useCallback, useState } from 'react'
 import { Trans, useTranslation } from 'react-i18next'
 import { DeletionModal } from '../../common/modals/deletion-modal'
 import { SidebarButton } from './sidebar-button'
 import type { SpecificSidebarEntryProps } from './types'
+import { useApplicationState } from '../../../hooks/common/use-application-state'
+import { cypressId } from '../../../utils/cypress-attribute'
+import { showErrorNotification } from '../../../redux/ui-notifications/methods'
+import { deleteNote } from '../../../api/notes'
 
 export const DeleteNoteSidebarEntry: React.FC<SpecificSidebarEntryProps> = ({ hide, className }) => {
   useTranslation()
   const [showDialog, setShowDialog] = useState(false)
+  const noteTitle = useApplicationState((state) => state.noteDetails.noteTitle)
+  const noteId = useApplicationState((state) => state.noteDetails.id)
+  const closeDialog = useCallback(() => setShowDialog(false), [setShowDialog])
+  const deleteNoteAndCloseDialog = useCallback(() => {
+    return deleteNote(noteId)
+      .catch(showErrorNotification('landing.history.error.deleteNote.text'))
+      .finally(() => {
+        setShowDialog(false)
+      })
+  }, [setShowDialog, noteId])
 
   return (
     <Fragment>
-      <SidebarButton icon={'trash'} className={className} hide={hide} onClick={() => setShowDialog(true)}>
+      <SidebarButton
+        {...cypressId('menu-delete')}
+        icon={'trash'}
+        className={className}
+        hide={hide}
+        onClick={() => setShowDialog(true)}>
         <Trans i18nKey={'landing.history.menu.deleteNote'} />
       </SidebarButton>
       <DeletionModal
-        onConfirm={() => setShowDialog(false)}
+        {...cypressId('deletionModal')}
+        onConfirm={deleteNoteAndCloseDialog}
         deletionButtonI18nKey={'editor.modal.deleteNote.button'}
         show={showDialog}
-        onHide={() => setShowDialog(false)}
+        onHide={closeDialog}
         title={'editor.modal.deleteNote.title'}>
         <h5>
           <Trans i18nKey={'editor.modal.deleteNote.question'} />
         </h5>
         <ul>
-          <li> noteTitle</li>
+          <li {...cypressId('deletion.noteTitle')}>&nbsp;{noteTitle}</li>
```

Comment:
Why the nbsp?

Reply:
Before it was a normal space, I think this is non-desirable therefore I replaced it with the nbsp